### PR TITLE
[FEATURE] DragGAN implementation with MMagic

### PR DESCRIPTION
**Pull request by qsun1**
## Motivation

Implementation of DragGAN with MMagic-StyleGAN2 engine.

## Modification
- [x] Enhacing the StyleGAN2 for the capability with the inference of DragGAN.
- [x] Implementation of Drag part, most of the code is borrowed from DragGAN official code.
- [x] Support the *Lionx512* checkpoint/demo that aligns from the provided official StyleGAN2 code and one configure file that corresponds to the ckpt.
- [x] Check the basic inference with the StyleGAN2 examples. 
- [x] Precommit code check with `precommit`.
- [x] Support more checkpoint in  *i.e.* elephants, dogs, cats-afhq, face-ffhq, horse.
- [ ] If needed, I would like to add more demo usage for StyleGAN2 due to a handful of changes in StyleGAN2 code. 

## Use cases 
Shown in `demo/demo_DragGAN/readme.md`, only 2 steps are needed for starting DragGAN demo: 
First, put your checkpoint path in `./checkpoints`, *e.g.* `./checkpoints/stylegan2_lions_512_pytorch_mmagic.pth`

Then, try on the script:

```
python demo/demo_DragGAN/gradio_draggan.py
```

### CODE REVIEW

**Comment by qsun1** [comment]
## Capability check with the previous usage of StyleGAN2
We check with such script:
```
import matplotlib.pyplot as plt
import mmcv
from mmagic.apis import MMagicInferencer

# Create a MMagicInferencer instance and infer
result_out_dir = '/pth/to/img'

editor = MMagicInferencer(
    'styleganv2',
    model_setting=0,
)
results = editor.infer(result_out_dir=result_out_dir)
print(results[0].keys())
img = mmcv.imread(result_out_dir)
plt.imshow(mmcv.bgr2rgb(img))
```

**Comment by qsun1** [comment]
## A short demo for Lionx512

https://github.com/open-mmlab/mmagic/assets/55343765/818569fb-ea55-48a2-9e5f-ebd5a669d918

File: .gitignore [hunk 1]
```diff
@@ -127,7 +127,7 @@ demo/*.png
 out/*.png
 tests/data/out
 resources/
-
+./my_code
```
Comment (liuwenran) [inline .gitignore:130]: better not to add this

**Comment by qsun1** [comment]
pipeline-draggan

**Comment by qsun1** [comment]
draggan-elephant1
draggan-elepant2

**Review by liuwenran — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**